libqtile/backend/wayland/xdgwindow.py

```diff
@@ -287,6 +284,17 @@ def place(
                 self._geom.height != geom.height,
             ]
         )
+        state = self.surface.toplevel._ptr.current
+        place_changed = any(
+            [
+                self.x != x,
+                self.y != y,
+                self._width != width,
+                self._height != height,
+                state.width != width,
+                state.height != height,
```

Comment:
Why do we need to check `width/height` and `_width/_height`?

Reply:
the state's width and height is the width and height what the window reports using wlroots APIs, whilst the self._width and self._height is the width and height of the window. If any of those are mismatched we need to resize the window. the check against `self._width` and `self._height` is potentially unneeded but maybe there is some edge case to consider

Reply:
OK. It's just not clear to me how they can be different. Doesn't `self.width` just retrieve `self._width`?

Reply:
It says `state.width` not `self.width`

Reply:
Ha ha. Wow. I obviously need to get my eyes tested!

Reply:
Haha it's fine, thanks for the review